routeros/provider_schema_helpers.go

```diff
@@ -606,12 +636,7 @@ var (
 	AlwaysPresentNotUserProvided = func(k, old, new string, d *schema.ResourceData) bool {
 		// For lists and sets, the key will look like `something.12345` or `something.#`.
 		// But in the raw config it will be just `something`.
-		k = strings.Split(k, ".")[0]
-
-		if old != "" && d.GetRawConfig().GetAttr(k).IsNull() {
-			return true
-		}
-		return false
+		return old != "" && isRawValueEmpty(d.GetRawConfig(), k)
```

Comment:
Maybe it makes sense not to make the check a separate function, since it is unlikely to be reused? And the logic of the decision will be located in one place.

Reply:
Well, my intent was to make it readable with the tailing recursion, but the helper itself seems a bit useless now.

I've refactored the helper to do the same without recursion as we get another set of parameters. It should still be readable 😀